### PR TITLE
Let GitHub generate sitemap.xml

### PR DESCRIPTION
**Pull request by claudiob**
According to [GitHub pages docs](https://help.github.com/articles/sitemaps-for-github-pages/), GitHub will automatically create (and keep up-to-date) a sitemap if:

- the 'github-pages' gem installed
- no 'sitemap.xml' file is present in the root folder

This approach is probably better than having to manually update the sitemap.xml using a [third-party plugin](https://github.com/recurser/jekyll-plugins) that has not been updated since 2013.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,3 +1,4 @@
 source 'https://rubygems.org'
 
 gem 'jekyll'
```
Comment (rafaelfranca) [inline Gemfile]: we can remove this I thik.

File: Gemfile [hunk 2]
```diff
@@ -1,3 +1,4 @@
...
+gem 'github-pages'
```
Comment (claudiob) [inline Gemfile]: Do you mean this line? Or the entire Gemfile?

File: Gemfile (hunk 1 above)
Comment (rafaelfranca) [inline Gemfile]: The gem `jekyll` since it is a dependecy of `github-pages`

Comment (claudiob) [inline Gemfile]: Good idea! Removed :scissors:

**Comment by claudiob** [comment]
Just to confirm the result of this PR: this was the content of the (manual) sitemap before this commit:

```xml
This XML file does not appear to have any style information associated with it. The document tree is shown below.
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
<url>
<loc>/community/</loc>
<lastmod>2014-09-08</lastmod>
</url>
<url>
<loc>/core/alumni/</loc>
<lastmod>2014-09-08</lastmod>
</url>
<url>
<loc>/core/</loc>
<lastmod>2014-09-08</lastmod>
</url>
<url>
<loc>/deploy/</loc>
<lastmod>2014-09-08</lastmod>
</url>
<url>
<loc>/documentation/</loc>
<lastmod>2014-09-08</lastmod>
</url>
<url>
<loc>/download/</loc>
<lastmod>2014-09-08</lastmod>
</url>
<url>
<loc>/</loc>
<lastmod>2014-09-08</lastmod>
</url>
<url>
<loc>/quotes/</loc>
<lastmod>2014-09-08</lastmod>
</url>
<url>
<loc>/screencasts/</loc>
<lastmod>2014-09-08</lastmod>
</url>
<url>
<loc>/security/</loc>
<lastmod>2014-09-08</lastmod>
</url>
</urlset>
```

and this is what the (automatically generated) sitemap looks like now:

```xml
This XML file does not appear to have any style information associated with it. The document tree is shown below.
<urlset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd">
<url>
<loc>http://rubyonrails.org/404.html</loc>
</url>
<url>
<loc>http://rubyonrails.org/core/</loc>
</url>
<url>
<loc>http://rubyonrails.org/</loc>
</url>
<url>
<loc>http://rubyonrails.org/deploy/</loc>
</url>
<url>
<loc>http://rubyonrails.org/download/</loc>
</url>
<url>
<loc>http://rubyonrails.org/screencasts/</loc>
</url>
<url>
<loc>http://rubyonrails.org/core/alumni/</loc>
</url>
<url>
<loc>http://rubyonrails.org/documentation/</loc>
</url>
<url>
<loc>http://rubyonrails.org/quotes/</loc>
</url>
<url>
<loc>http://rubyonrails.org/community/</loc>
</url>
<url>
<loc>http://rubyonrails.org/security/</loc>
</url>
</urlset>
```

**Pull request merged**